account_chart_update/wizard/wizard_chart_update.py

```diff
@@ -483,8 +581,13 @@ def diff_fields(self, template, real):
                 if expected is not None:
                     if expected != [] and expected != real[key]:
                         result[key] = expected
-                elif template[key] != real[key]:
-                    result[key] = template[key]
+                else:
+                    template_value = template[key]
+                    if template._name == "account.account.template" \
+                            and key == 'code':
+                        template_value = self.padded_code(template['code'])
```

Comment:
Why do we need this now and not before?

Reply:
Previously, the matching was only with the code field, so when an account (account.account) matching with the template (account.account.template) it was because the code was the same, therefore, it was not necessary to verify if they were different or not. (It wasn't in tab *field options > Accounts*). Now the matching can be done with the XML- ID or with another field different from the code, so now we can verify if the code changes, that's why I've also added it to the *Field Options* tab. But, to know if it changed, we must pad with zeros to the right, because the code field is not padded with zeros in the template but it's padded with zeros in the account. This is the same that it was done to match by code before making this module: 
https://github.com/OCA/account-financial-tools/blob/94ccc6987a6f0a23c01d0819043dfa09c0508ed3/account_chart_update/wizard/wizard_chart_update.py#L337

Reply:
OK, got it.